tower-grpc-interop/src/client.rs

```diff
@@ -216,6 +216,302 @@ fn assert_success(result: Result<Vec<TestAssertion>, Box<Error>>)
     })
 }
 
+type TestClient = pb::client::TestService<tower_http::AddOrigin<
+    tower_h2::client::Connection<
+        tokio_core::net::TcpStream, tokio_core::reactor::Handle, tower_h2::BoxBody>>>;
+
+fn empty_unary_test(client: &mut TestClient, core: &mut tokio_core::reactor::Core)
+        -> Result<Vec<TestAssertion>, Box<Error>> {
+    use pb::Empty;
+    core.run(client
```

Comment:
As a side note: when I was writing this code originally, my initial intention was to have functions for each test that returned `Future<Item = Vec<TestAssertion>, Error = Box<Error>>` that we could then run on a `core` in the `main` function.

At the time, I ended up not doing that, since the returned futures weren't nameable types (as they contained closures), and it seemed like too much effort would be required to get around that. Now, I think we could do that pretty easily, with `impl Future`. 

This isn't a change you have to make in _this_ branch, but in the long run, I think it might be a nicer way to structure this code. Perhaps something that either of us could do in a follow-up branch...

Reply:
That is nicer. Fixed.